### DISCUSSION TITLE
Parameter conversion and set attribute "inconsistency"

### DISCUSSION THREAD

**Discussion by StefanoTesla** [mathieucarbou/ESPAsyncWebServer#174] [Ideas]
When I read a parameter, i can convert it, that's nice!

![image](https://github.com/user-attachments/assets/55299668-5787-4568-a89b-de67878c41b4)

But, if I want to pass it as request attribute I have others options:

![image](https://github.com/user-attachments/assets/bfef7788-e539-47cb-ab86-b386b6a56a09)

Can you consider to add int on the setParameters input value?

The work around is little adding a static_cast<long> ex: request->setAttribute("hello", static_cast<long>(10));

**Comment by StefanoTesla** [comment]
probably doesen't care, since getAttribute return always a String...

**Reply by mathieucarbou** [reply]
yes they were added for convenience purposes. Also, `toInt()` is from `String` and returns a long, not an int. You don't see it because of the implicit conversion. int and long are both on 4 bytes on ESP32.

int theory, you should better use int32_t, uint32_t, int16_t, uint16_t, etc and not use toInt(), toFloat() etc.

`String` class was mostly introduced because STL (std::string) usage is not possible on AVR. There is no point in using String and not using std::string on ESP32.
